core/src/apps/cardano/sign_tx.py

```diff
@@ -61,18 +75,34 @@ async def sign_tx(
     return tx
 
 
+def _validate_network_info(network_id: int, protocol_magic: int) -> None:
+    is_mainnet_network_id = network_ids.is_mainnet(network_id)
+    is_mainnet_protocol_magic = protocol_magics.is_mainnet(protocol_magic)
+
+    if is_mainnet_network_id != is_mainnet_protocol_magic:
+        raise wire.ProcessError("Invalid network id/protocol magic combination!")
+
+
 def _validate_outputs(
-    keychain: seed.Keychain, outputs: List[CardanoTxOutputType], protocol_magic: int
+    keychain: seed.Keychain,
+    outputs: List[CardanoTxOutputType],
+    protocol_magic: int,
+    network_id: int,
 ) -> None:
+    if len(outputs) < 1:
+        raise wire.ProcessError("Transaction has no outputs!")
+
     total_amount = 0
     for output in outputs:
         total_amount += output.amount
-        if output.address_n:
+        if output.address_parameters:
```

Comment:
istm the address parameters should be validated here as well

Reply:
I've done this [here](https://github.com/trezor/trezor-firmware/pull/1112/commits/02d0646b9e71a3936fe2627c81ef77f5377bf791). However, I did it by simply trying to derive the address, since there is no explicit validation for this and I couldn't find a way of validating the address without duplicating too much logic. Please let me know what you think of this.

Reply:
that works, I suppose.
you don't need the `_ =` assignment though

Reply:
Ok. Done [here](https://github.com/trezor/trezor-firmware/pull/1112/commits/df61674ca55f4e03f1042a797e39ad0d7c07c4c5).